packages/@uppy/react/src/useUppyState.js

```diff
@@ -0,0 +1,20 @@
+import { useSyncExternalStore, useCallback } from 'react'
+
+export default function useUppyState (uppy, selector) {
+  // Eslint complains the dependencies are unknown and we should use an inline function
+  // but we put uppy.store.subscribe in there ourselves.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const subscribe = useCallback(uppy.store.subscribe.bind(uppy.store), [
+    uppy.store.subscribe,
+  ])
+  const getSnapshot = useCallback(
+    () => selector(uppy.store.getState()),
+    // Eslint wants to put `uppy.store` as the dependency
+    // but that seems unnecessary as we can be more specific.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [uppy.store.getState, selector],
+  )
```

Comment:
@nickrttn am I committing acceptable React crimes here or no?

Reply:
what about instead `const subscribe = useMemo(() => uppy.store.subscribe.bind(uppy.store), [uppy.store])` then we don't need to eslint disable and we don't risk crash if uppy.store happened to be undefined

Reply:
☝️ what @mifi says 👍

Reply:
Your suggestion would need to be changed into this:

```js
  const subscribe = useMemo(
    (listener) => uppy.store.subscribe.bind(uppy.store)(listener),
    [uppy.store],
  )
```

which does crashes at runtime. Changing it to `useCallback` does work. But then might as well keep what I have here?